olga/analytics/models.py

```diff
@@ -149,3 +155,98 @@ def worlds_students_per_country_statistics(cls):  # pylint: disable=invalid-name
+        """
+        students_amount_percentage = format(
+            country_count_in_statistics / all_active_students * 100, '.2f'
+        )
+
+        if students_amount_percentage == '0.00':
+            students_amount_percentage = '~0'
+
+        return students_amount_percentage
+
+    @staticmethod
+    def does_country_exists(country):
+        """
+        Check if value is not a null.
+        """
+        return country != 'null'
+
+    @classmethod
+    def create_students_per_country_to_render(cls, worlds_students_per_country):
+        # pylint: disable=invalid-name
+        """
+        Create convenient and necessary data formats to render it from view.
+
+        Graphs require list-format data.
+        """
+        datamap_format_countries_list = []
+        tabular_format_countries_list = []
+
+        if not worlds_students_per_country:
+            tabular_format_countries_list.append(('Unset', 0, 0))
+            return datamap_format_countries_list, tabular_format_countries_list
+
+        all_active_students = sum(worlds_students_per_country.itervalues())
+
+        for country, count in worlds_students_per_country.iteritems():
+            student_amount_percentage = cls.get_student_amount_percentage(count, all_active_students)
+
+            if cls.does_country_exists(country):
+                country = str(pycountry.countries.get(alpha_2=country).alpha_3)
+
+                datamap_format_countries_list += [[country, count]]
+                tabular_format_countries_list += [[country, count, student_amount_percentage]]
+
+            else:
+                # Create students without country amount.
+                tabular_format_countries_list += [['Unset', count, student_amount_percentage]]
+
+        # Sort in descending order.
+        tabular_format_countries_list.sort(key=lambda row: row[1], reverse=True)
+
+        return datamap_format_countries_list, tabular_format_countries_list
+
+    @classmethod
+    def get_students_per_country_to_render(cls):
+        # pylint: disable=invalid-name
+        """
+        Gather convenient and necessary data formats to render it from view.
+        """
+        students_per_country = cls.get_students_per_country_stats()
+
+        datamap_format_countries_list, tabular_format_countries_list = \
+            cls.create_students_per_country_to_render(students_per_country)
+
+        return datamap_format_countries_list, tabular_format_countries_list
+
+    @staticmethod
+    def calculate_countries_amount(tabular_format_countries_list):
+        """
+        Calculate countries amount in worlds students per country statistics as table.
+
+        If no countries - appends `Unset` field with zero countries, else get.
+        Actually `Unset` field is not a country, so it does not fill up in countries amount.
+        """
+        if not tabular_format_countries_list:
+            return 0
+
+        return len(tabular_format_countries_list) - 1
```

Comment:
From method docstring: `Actually 'Unset' field is not a country, so it does not fill up in countries amount.`. I'll extend documentation.